src/main/java/traincraft/blocks/assemblytables/GuiAssemblyTable.java

```diff
@@ -0,0 +1,132 @@
+    private static final ResourceLocation GUI_LOCATION_TIER2 = new ResourceLocation(Traincraft.MOD_ID, "textures/gui/gui_assemblytable_tier2.png");
+    private static final ResourceLocation GUI_LOCATION_TIER3 = new ResourceLocation(Traincraft.MOD_ID, "textures/gui/gui_assemblytable_tier3.png");
+
+    /*
+     * Locations (pixels) of GUI elements. From top left corner, first pixel that should be in image.
+     * Width, Height of GUI: (176, 256)
+     *
+     * Red Border (178, 1); (32, 32) size
+     * Green Border (209, 1); (32, 32) size
+     *
+     * Red strong border: (181, 63); (24, 24) size
+     * Green Strong Border (212, 63); (24, 24) size
+     */
+    public GuiAssemblyTable(int tier, EntityPlayer player, TileAssemblyTable tileAssemblyTable) {
+        super(new ContainerAssemblyTable(player.inventory, player.getEntityWorld(), tileAssemblyTable));
+        this.tier = tier;
+        this.player = player;
+        this.tileAssemblyTable = tileAssemblyTable;
+    }
+    
+    @Override
+    public void drawScreen(int mouseX, int mouseY, float partialTicks){
+        this.drawDefaultBackground();
+        super.drawScreen(mouseX, mouseY, partialTicks);
+        this.renderHoveredToolTip(mouseX, mouseY);
+    }
+    
+    @Override
+    protected void drawGuiContainerForegroundLayer(int mouseX, int mouseY) {
+    }
+
+    @Override
+    protected void drawGuiContainerBackgroundLayer(float partialTicks, int mouseX, int mouseY) {
+        GlStateManager.color(1.0F, 1.0F, 1.0F, 1.0F);
+
+        //draw the gui image based on location
+        ResourceLocation GUI_LOC;
+        switch (tier) {
+            case 1:
+                GUI_LOC = GUI_LOCATION_TIER1;
+                break;
+            case 2:
+                GUI_LOC = GUI_LOCATION_TIER2;
+                break;
+            case 3:
+                GUI_LOC = GUI_LOCATION_TIER3;
+                break;
+            default:
+                Traincraft.LOGGER.log(Level.ERROR, "Invalid tier found when opening Assembly Table Gui.");
+                //set to the Tier1, just so things don't fall apart.
+                GUI_LOC = GUI_LOCATION_TIER1;
+                break;
+        }
+
+        this.mc.renderEngine.bindTexture(GUI_LOC);
+        this.drawTexturedModalRect((width - 176) / 2, (height - 256) / 2, 0, 0, 176, 256);
+
+        //draw the text
+        //it seems that minecraft is not prepared to have different size guis than in vanilla, so we must do manual interpolation
+        int adjustedX = (this.width - 176) / 2;
+        int adjustedY = (this.height - 256) / 2;
+
+        //TODO: are these colors ok?
+        this.fontRenderer.drawString(I18n.format(Traincraft.MOD_ID + ":assembly_table_" + tier + ".name"), 10 + adjustedX, 6 + adjustedY, new Color(186, 201, 48).getRGB());
+
+        int color = new Color(147, 145, 158).getRGB();
```

Comment:
Doing this every frame tick would be bad. This has to be a fixed hex integer value, which you can just calculate in a online converter and paste it here, instead of calculating it every frame

Reply:
agreed, fixed